python/pyspark/ml/clustering.py

```diff
@@ -167,6 +167,200 @@ def getInitSteps(self):
+    >>> model = lda.fit(df)
+    >>> model.isDistributed()
+    True
+    >>> localModel = model.toLocal()
+    >>> localModel.isDistributed()
+    False
+    >>> model.vocabSize()
+    2
+    >>> model.describeTopics().show()
+    +-----+-----------+--------------------+
+    |topic|termIndices|         termWeights|
+    +-----+-----------+--------------------+
+    |    0|     [1, 0]|[0.50401530077160...|
+    |    1|     [0, 1]|[0.50401530077160...|
+    +-----+-----------+--------------------+
+    ...
+    >>> model.topicsMatrix()
+    DenseMatrix(2, 2, [0.496, 0.504, 0.504, 0.496], 0)
+
+    .. versionadded:: 1.7.0
+    """
+
+    # a placeholder to make it appear in the generated doc
+    k = Param(Params._dummy(), "k", "number of clusters to create")
+    optimizer = Param(Params._dummy(), "optimizer", "LDA optimizer")
+
+    @keyword_only
+    def __init__(self, featuresCol="features", k=2,
+                 optimizer="online", learningOffset=5, learningDecay=0.51,
+                 subsamplingRate=0.05, optimizeDocConcentration=True,
+                 checkpointInterval=10, maxIter=20, seed=None):
+        """
+        __init__(self, featuresCol="features", predictionCol="prediction", k=2, \
+                 initMode="k-means||", initSteps=5, tol=1e-4, maxIter=20, seed=None)
+        """
+        super(LDA, self).__init__()
+        self._java_obj = self._new_java_obj("org.apache.spark.ml.clustering.LDA", self.uid)
+        self.k = Param(self, "k", "number of clusters to create")
+        self.optimizer = Param(self, "optimizer", "LDA optimizer")
+        self._setDefault(k=2, optimizer="online", maxIter=20)
+        kwargs = self.__init__._input_kwargs
+        self.setParams(**kwargs)
+
+    def _create_model(self, java_model):
+        if self.getOptimizer() == "em":
+            return DistributedLDAModel(java_model)
+        else:
+            return LocalLDAModel(java_model)
+
+    @keyword_only
+    @since("1.7.0")
+    def setParams(self, featuresCol="features", k=2,
+                  optimizer="online", learningOffset=5, learningDecay=0.51,
+                  subsamplingRate=0.05, optimizeDocConcentration=True,
+                  checkpointInterval=10, maxIter=20, seed=None):
+        """
+        ssetParams(self, featuresCol="features", k=2,
+                  optimizer="online", learningOffset=5, learningDecay=0.51,
+                  subsamplingRate=0.05, optimizeDocConcentration=True,
+                  checkpointInterval=10, maxIter=20, seed=None):
+
+        Sets params for LDA.
+        """
+        kwargs = self.setParams._input_kwargs
+        return self._set(**kwargs)
+
+    @since("1.7.0")
+    def setK(self, value):
+        """
+        Sets the value of :py:attr:`k`.
+
+        >>> algo = LDA().setK(10)
+        >>> algo.getK()
+        10
+        """
+        self._paramMap[self.k] = value
+        return self
+
+    @since("1.7.0")
+    def getK(self):
+        """
+        Gets the value of `k`
```

Comment:
nit: ```Gets the value of k or its default value.```
I guess you refer python/pyspark/ml/clustering.py, maybe we should also update that class.